src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -98,7 +98,7 @@ public static function getSubscribedEvents()
             Events::HYDRATE => ['handleHydrate', -200],
             Events::MOVE => ['updateMovedDocument', -128],
             Events::COPY => ['updateCopiedDocument', -128],
-            Events::PUBLISH => 'handlePersistRoute',
+            Events::PUBLISH => ['handlePersistRoute', -128],
```

Comment:
because if a page gets published, the workflow-stage must be updated before the route is saved and therefore the children of the page get adapted.
otherwise, if the page was unpublished before, the children will ignore the resource locator of the parent because the parent lstates he is unpublished at that moment.